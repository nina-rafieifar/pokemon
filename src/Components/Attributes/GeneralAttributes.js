import { html, LitElement, css } from "lit-element";

class PokemonAttr extends LitElement {
  static get properties() {
    return {
      name: { attribute: true },
      number: { attribute: true },
      image: { attribute: true }
    };
  }
  static get styles() {
    return css`
      .pokemon__general-attributes {
        box-shadow: 0 16px 32px 0 rgba(0, 0, 0, 0.2);
        transition: 0.3s;
        border-radius: 22px;
        background-color: #ffffff;
        max-height: 87vh;
      }
      .pokemon__general-attributes:hover {
        box-shadow: 0 35px 45px 0 rgba(0, 0, 0, 0.2);
      }
      .pokemon__name {
        margin: 0;
      }
      .pokemon__number {
        padding: 1rem 0;
        font-family: Tigerious;
      }
      .pokemon__image {
        height: 50vh;
        pointer-events: none;
        text-align: left;
      }
    `;
  }
  render() {
    return html`<div class="pokemon__general-attributes">
      <h2 class="pokemon__number">${this.number}</h2>
      <img alt=${this.name} src=${this.image} class="pokemon__image" />
      <h1 class="pokemon__name">${this.name}</h1>
    </div>`;
  }
}

customElements.define("pokemon-general-attr", PokemonAttr);
