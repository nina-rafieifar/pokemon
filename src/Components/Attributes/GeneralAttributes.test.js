import "./GeneralAttributes";
import { fixture } from "@open-wc/testing";
import { html, render } from "lit-html";

test("pokemon-general-attr renders the correct elements", () => {
  render(
    html`<pokemon-general-attr name="Ivysaur" number="002" image="some url"></pokemon-general-attr>`,
    document.body
  );
  expect(document.body.getElementsByTagName("pokemon-general-attr")).toBeDefined();
});

test("pokemon-general-attr renders as it should", async () => {
  const element = await fixture(
    html`<pokemon-general-attr name="Ivysaur" number="002" image="some url"></pokemon-general-attr>`
  );
  expect(element).toMatchSnapshot();
});
