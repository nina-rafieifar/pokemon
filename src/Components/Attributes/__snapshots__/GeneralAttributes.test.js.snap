// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemon-general-attr renders as it should 1`] = `
<pokemon-general-attr
  image="some url"
  name="Ivysaur"
  number="002"
/>
`;
